application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
The XHR implementation seemed solid to me.  The http library just loses the preflight (OPTIONS) response.